apps/native-ui-storybook/components/List/List.docs.mdx

```diff
@@ -0,0 +1,660 @@
+        <ListItemLeadingContent>
+          <Box padding="$3" borderRadius="$lg" bg="$apple50">
+            <ListItemIcon as={GasMediumIcon} color="$apple800" />
+          </Box>
+        </ListItemLeadingContent>
+        <ListItemContent>
+          <HStack justifyContent="space-between">
+            <ListItemText>Electricity</ListItemText>
+            <Badge colorScheme="green" size="small">
+              <BadgeText>Smart meter</BadgeText>
+            </Badge>
+          </HStack>
+          <ListItemSupportingText>Last reading 23/03/24</ListItemSupportingText>
+        </ListItemContent>
+        <ListItemTrailingContent>
+          <ListItemTrailingIcon as={ChevronRight01MediumIcon} />
+        </ListItemTrailingContent>
+      </ListItem>
+    </List>
+  </NativeUIProvider>
+</Canvas>
+
+```tsx
+import {
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemLeadingContent,
+  ListItemSupportingText,
+  ListItemText,
+  ListItemTrailingContent,
+  ListItemTrailingIcon,
+  ListItemContent,
+  Badge,
+  BadgeText,
+  Box,
+  HStack,
+} from '@utilitywarehouse/native-ui';
+import { ElectricityMediumIcon, GasMediumIcon } from '@utilitywarehouse/react-native-icons';
+
+const MyComponent = () => (
+  <List>
+    <ListItem onPress={() => console.log('pressed')} divider>
+      <ListItemLeadingContent>
+        <Box padding="$3" borderRadius="$lg" bg="$apple50">
+          <ListItemIcon as={ElectricityMediumIcon} color="$apple800" />
+        </Box>
+      </ListItemLeadingContent>
+      <ListItemContent>
+        <ListItemText>Electricity</ListItemText>
+        <Badge colorScheme="cyan" size="small">
+          <BadgeText>Text</BadgeText>
+        </Badge>
+        <ListItemSupportingText>Last reading 23/03/24</ListItemSupportingText>
+      </ListItemContent>
+      <ListItemTrailingContent>
+        <ListItemTrailingIcon as={ChevronRight01MediumIcon} />
+      </ListItemTrailingContent>
+    </ListItem>
+    <ListItem onPress={() => console.log('pressed')} divider={false}>
+      <ListItemLeadingContent>
+        <Box padding="$3" borderRadius="$lg" bg="$apple50">
+          <ListItemIcon as={GasMediumIcon} color="$apple800" />
+        </Box>
+      </ListItemLeadingContent>
+      <ListItemContent>
+        <HStack justifyContent="space-between">
+          <ListItemText>Electricity</ListItemText>
+          <Badge colorScheme="green" size="small">
+            <BadgeText>Smart meter</BadgeText>
+          </Badge>
+        </HStack>
+        <ListItemSupportingText>Last reading 23/03/24</ListItemSupportingText>
+      </ListItemContent>
+      <ListItemTrailingContent>
+        <ListItemTrailingIcon as={ChevronRight01MediumIcon} />
+      </ListItemTrailingContent>
+    </ListItem>
+  </List>
+);
+```
+
+### Checbox List
+
+You can also use the list to display a list of checkboxes. You can achieve this by using the `Checkbox` component as the `trailingContent` of the `ListItem`.
```

Comment:
Can we mirror what we have in Figma and also offer an example with Switch and Radio?

Reply:
We don't have the switch component yet but we can add the radio yes